src/scripts/ui/ConstructionPage.tsx

```diff
@@ -48,9 +57,20 @@ export function ConstructionPage({ tile }: { tile: ITileData }): React.ReactNode
          notifyGameStateUpdate();
       }
    };
+
+   const levels = getBuildingUpgradeLevels(building);
+   const upgrade = (level: number) => {
+      building.desiredLevel = building.desiredLevel + level;
+      building.status = "upgrading";
+      notifyGameStateUpdate();
+   };
+
    useShortcut("UpgradePageIncreaseLevel", () => increaseDesiredLevel(), [tile]);
    useShortcut("UpgradePageDecreaseLevel", () => decreaseDesiredLevel(), [tile]);
 
+   useShortcut("BuildingPageUpgradeX5", () => upgrade(levels[0]), [tile]);
```

Comment:
Shortcut in the game are scoped to avoid conflict. This shortcut only applies to the building page. For upgrade page, you need a new scope